cybox/objects/artifact_object.py

```diff
@@ -23,11 +23,27 @@ def validate_artifact_type(instance, value):
         raise ValueError(err)
 
 
+def validate_byte_order_endianness(instance, value):
+    if value is None:
+        return
+    elif value in RawArtifact.ENDIANNESS:
+        return
+    else:
+        err = "Type must be one of %s. Received '%s'." % (RawArtifact.ENDIANNESS, value)
+        raise ValueError(err)
+
+
 class RawArtifact(String):
-    _binding_class = artifact_binding.RawArtifactType
+    _binding = artifact_binding
+    _binding_class = _binding.RawArtifactType
     _namespace = 'http://cybox.mitre.org/objects#ArtifactObject-2'
 
-    byte_order = fields.TypedField("byte_order")
+    BIG_ENDIAN = "Big-endian"
+    LITTLE_ENDIAN = "Little-endian"
+    MIDDLE_ENDIAN = "Middle-endian"
+    ENDIANNESS = (BIG_ENDIAN, LITTLE_ENDIAN, MIDDLE_ENDIAN)
+
+    byte_order = fields.TypedField("byte_order", postset_hook=validate_byte_order_endianness)
```

Comment:
`validate_byte_order_endianness` can probably be a pre-set hook, since it doesn't depend on any of the other values of the instance.